### PR TITLE
[DPE-4120] Add e2e test

### PR DESCRIPTION
**Pull request by welpaolo**

### CODE REVIEW

File: python/tests/integration/test_spark_job.py [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+            "--username",
+            service_account.name,
+            "--namespace",
+            service_account.namespace,
+            "-v",
+            "--class",
+            "org.apache.spark.examples.SparkPi",
+            "local:///opt/spark/examples/jars/spark-examples_2.12-3.4.2.jar",
+            "1000",
+        ]
```
Comment (deusebio) [inline python/tests/integration/test_spark_job.py:205]: **todo** It would be nice to use a job that reads from S3

**Review by deusebio — COMMENTED** [review]
Looks great! If it is reasonably easy, I would just make it such we also read data from s3, to really make it a end-to-end tests

File: python/tests/integration/test_spark_job.py (hunk 1 above)
Comment (welpaolo) [reply python/tests/integration/test_spark_job.py:133]: Done

File: python/spark_test/core/pod.py [hunk 2]
```diff
@@ -138,6 +142,22 @@ def logs(self) -> Iterator[str]:
...
 
+    def metadata(self):
+        """Return the metadata of the pod."""
+        return self.client.get(
+            lightkube.resources.core_v1.Pod,
+            name=self.pod_name,
+            namespace=self.namespace,
+        ).metadata
+
+    def labels(self) -> Dict[str, str]:
```
[the remark this answers is not in the document]
Comment (welpaolo) [reply python/spark_test/core/pod.py:153]: Done

File: .github/workflows/ci-tests.yaml [hunk 3]
```diff
@@ -108,4 +112,4 @@ jobs:
       - id: tests-integration
         name: Run Integration Tests
         run: |    
-          cd python && tox run -e integration -- ${{ matrix.backend }} ${{ matrix.cos }}
\ No newline at end of file
+          cd python && tox run -e ${{ matrix.tox-environments }} -- -m '${{ steps.select-tests.outputs.mark_expression }}' ${{ matrix.backend }} ${{ matrix.options }}
```
Comment (deusebio) [inline .github/workflows/ci-tests.yaml:115]: shouldn't this be `matrix.cos`?

**Review by deusebio — APPROVED** [review]
LGTM. Only a small detail, I suppose

File: .github/workflows/ci-tests.yaml (hunk 3 above)
Comment (welpaolo) [reply .github/workflows/ci-tests.yaml]: Done

File: python/spark_test/core/pod.py [hunk 4]
```diff
@@ -142,6 +142,24 @@ def logs(self) -> Iterator[str]:
...
+        ).metadata
+
+    @property
+    def labels(self) -> Dict[str, str]:
+        """Return the labels of the pod."""
+        return self.client.get(
+            lightkube.resources.core_v1.Pod,
+            name=self.pod_name,
+            namespace=self.namespace,
+        ).metadata.labels
```
Comment (theoctober19th) [inline python/spark_test/core/pod.py:161]: Not something significant, but this could also be shortened to `return self.metadata.labels`.

**Review by theoctober19th — APPROVED** [review]
LGTM, great work!

**Pull request merged**